core/services/mongodb-remote/src/main/java/com/mongodb/stitch/core/services/mongodb/remote/sync/internal/SyncWriteModelContainer.java

```diff
@@ -69,7 +69,7 @@ void wrapForRecovery(final MongoCollection<BsonDocument> localCollection,
                        final MongoCollection<BsonDocument> undoCollection,
                        final Runnable callable) {
     final List<BsonDocument> oldDocs = localCollection.find(
-        new Document("_id", new Document("$in", ids))
+        new BsonDocument("_id", new BsonDocument("$in", new BsonArray(new ArrayList<>(ids))))
```

Comment:
[q] just curious, what was this drive-by for?

Reply:
I was testing to see if maybe there was a weird overhead in adding a `Set` to a `Document`. I can change it back, but I don't think it matters either way.